### PR TITLE
Casmmon 111 snmp-exporter documentation

### PR DESCRIPTION
**Pull request by rambabubolla**
## Summary and Scope

Change order to update Prometheus SNMP Exporter Deployment values.yaml

## Issues and Related PRs
* Resolves [iCASMMON-111](https://jira-pro.its.hpecorp.net:8443/browse/CASMMON-111)
*

## Testing

No new code, just documentation

### Test description:
-NA

### CODE REVIEW

File: operations/network/management_network/snmp_exporter_configs.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+For a complete set of available settings, consult the values.yaml file for the `cray-sysmgmt-health` chart. The most common customizations to set are specified in the following table. They must be set in the customizations.yaml file under the `prometheus-snmp-exporter:` setting.
+
+|Customization|Default|Description|
+|-------------|-------|-----------|
+|`serviceMonitor.enabled`|`true`|Enables serviceMonitor for snmp exporter \(default chart value is `true`\)|
+|`params.enabled`|`false`|Sets the snmp exporter params change to true \(default chart value is `false`\)|
+|`params.conf.module`|`if_mib`| snmp exporter to select which module \(default chart value is `if_mib`\)|
+|`params.conf.target`|`127.0.0.1`| add list of switch targets to snmp exporter to monitor \(default chart value is `127.0.0.1`\)|
+
+#### Obtain the list of switches to as targets
```
[the remark this answers is not in the document]
Comment (spillerc-hpe) [reply operations/network/management_network/snmp_exporter_configs.md:15]: There isn't really a nice way of getting the list of switch from the `csi` output files at install time without looking at `/var/www/ephemeral/prep/${SYSTEM_NAME}/sls_input_file.json`

CANU maybe able to help here as it consumes that file along with the SHCD to build the switch configs, just need to figure out the correct point in the install workflow to drop this in.

[the remark this answers is not in the document]
Comment (spillerc-hpe) [reply operations/network/management_network/snmp_exporter_configs.md:15]: It looks like CANU will fit the bill here
```
wasp-ncn-m001-pit:~ # canu init --sls-file /var/www/ephemeral/prep/wasp/sls_input_file.json --out -
10.252.0.2
10.252.0.3
10.252.0.4
10.252.0.5
4 IP addresses saved to <stdout>
```

**Review by spillerc-hpe — CHANGES_REQUESTED** [review]

File: operations/network/management_network/snmp_exporter_configs.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+## Prometheus SNMP Exporter Deployment
+rometheus SNMP Exporter is deployed with the `cray-sysmgmt-health` chart to the `sysmgmt-health` namespace as part of the Cray System Management \(CSM\) release.
+
+### Customizations
+
+For a complete set of available settings, consult the values.yaml file for the `cray-sysmgmt-health` chart. The most common customizations to set are specified in the following table. They must be set in the customizations.yaml file under the `prometheus-snmp-exporter:` setting.
```
[the remark this answers is not in the document]
Comment (rambabubolla) [reply operations/network/management_network/snmp_exporter_configs.md:6]: done

File: operations/network/management_network/snmp_exporter_configs.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rambabubolla) [reply operations/network/management_network/snmp_exporter_configs.md]: done

File: operations/network/management_network/snmp_exporter_configs.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+## Prometheus SNMP Exporter Deployment
+rometheus SNMP Exporter is deployed with the `cray-sysmgmt-health` chart to the `sysmgmt-health` namespace as part of the Cray System Management \(CSM\) release.
```
Comment (nrockershousen) [inline operations/network/management_network/snmp_exporter_configs.md:2]: "rometheus" should be "Prometheus"

File: operations/network/management_network/snmp_exporter_configs.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+For a complete set of available settings, consult the values.yaml file for the `cray-sysmgmt-health` chart. The most common customizations to set are specified in the following table. They must be set in the customizations.yaml file under the `prometheus-snmp-exporter:` setting.
+
+|Customization|Default|Description|
+|-------------|-------|-----------|
+|`serviceMonitor.enabled`|`true`|Enables serviceMonitor for snmp exporter \(default chart value is `true`\)|
+|`params.enabled`|`false`|Sets the snmp exporter params change to true \(default chart value is `false`\)|
+|`params.conf.module`|`if_mib`| snmp exporter to select which module \(default chart value is `if_mib`\)|
+|`params.conf.target`|`127.0.0.1`| add list of switch targets to snmp exporter to monitor \(default chart value is `127.0.0.1`\)|
+
+prometheus-snmp-exporter block will look like below in customizations.yaml
```
Comment (nrockershousen) [inline operations/network/management_network/snmp_exporter_configs.md:15]: Change to:

The prometheus-snmp-exporter block in the customizations.yaml file will look similar to the following:

File: operations/network/management_network/snmp_exporter_configs.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+                target:
+                - 127.0.0.1
+
+#### Obtain the list of switches to as targets
+
+From the command line to run:
+Example:
+
+```bash
+wasp-ncn-m001-pit:~ # canu init --sls-file /var/www/ephemeral/prep/wasp/sls_input_file.json --out -
```
Comment (nrockershousen) [inline operations/network/management_network/snmp_exporter_configs.md:35]: Command prompt should be updated to follow docs-csm conventions and remove internal system name. It should be:

ncn-m001-pit#

File: operations/network/management_network/snmp_exporter_configs.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+10.252.0.2
+10.252.0.3
+10.252.0.4
+10.252.0.5
+4 IP addresses saved to <stdout>
```
Comment (nrockershousen) [inline operations/network/management_network/snmp_exporter_configs.md:40]: Missing ``` at the end of this code block

File: operations/network/management_network/snmp_exporter_configs.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+            enabled: true
+            params:
+              enabled: true
+              conf:
+                module:
+                - if_mib
+                target:
+                - 127.0.0.1
+
+#### Obtain the list of switches to as targets
```
Comment (nrockershousen) [inline operations/network/management_network/snmp_exporter_configs.md:29]: Looks like a word is missing. Change to:

Obtain the list of switches to use as targets

File: operations/network/management_network/snmp_exporter_configs.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+From the command line to run:
```
Comment (nrockershousen) [inline operations/network/management_network/snmp_exporter_configs.md:31]: Change lines 31/32:

The following is an example of collecting a list of switches:

File: operations/network/management_network/snmp_exporter_configs.md (hunk 3 above)
Comment (rambabubolla) [reply operations/network/management_network/snmp_exporter_configs.md]: updated

File: operations/network/management_network/snmp_exporter_configs.md (hunk 4 above)
Comment (rambabubolla) [reply operations/network/management_network/snmp_exporter_configs.md]: updated

File: operations/network/management_network/snmp_exporter_configs.md (hunk 5 above)
Comment (rambabubolla) [reply operations/network/management_network/snmp_exporter_configs.md]: updated

**Review by spillerc-hpe — APPROVED** [review]
I have made a few changes and reworked the procedure to be consistent with the `prepare_site_init.md` document that this procedure is linked from.

File: operations/network/management_network/snmp_exporter_configs.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+In order to provide data to the Grafana SNMP dashboards the SNMP Exporter must be configured with a list of management network switches to scrape metrics from.
+
+1. Set the `SYSTEM_NAME` variable if not already set.
+
+```bash
+linux# SYSTEM_NAME=eniac
+```
+
+1. Obtain the list of switches to use as targets using CANU (CSM Automatic Network Utility)
```
Comment (nrockershousen) [inline operations/network/management_network/snmp_exporter_configs.md:14]: - All of the steps below are missing a period at the end of the line. Please add one to each step.
- The spelled out acronym name should be listed first: CSM Automatic Network Utility (CANU)

Comment (rambabubolla) [reply operations/network/management_network/snmp_exporter_configs.md]: done

**Review by nrockershousen — APPROVED** [review]

**Pull request merged**